i6_models/config.py

```diff
@@ -51,3 +53,18 @@ def _validate_types(self) -> None:
 
     def __post_init__(self) -> None:
         self._validate_types()
+
+
+@dataclass
+class SubassemblyWithOptions:
+    """
+    Dataclass for a combination of an Subassembly/Part and the corresponding configuration.
+    Also provides a function to construct the corresponding object through this dataclass
+    """
+
+    module_class: Callable[[ModelConfiguration], nn.Module]
```

Comment:
```suggestion
    module_class: type
```
In my opinion expecting a callable here makes this class a little bit too complicated to use. I would expect it to work simply with something like
`frontend = SubassemblyWithOptions(ConformerVGGFrontendV1, frontend_cfg)`.
and not have to introduce some lambda function or whatever to get a callable.
You could get fancy and introduce some abstract base class that all parts/assemblies derive from to make sure that the constructor actually works with an instance of `ModelConfiguration`, but I think simply annotating with `type` is fine.

Reply:
Hm yes, it seems that you are right. On my side Pylance raised the issue
`Argument of type "Type[ConformerVGGFrontendV1]" cannot be assigned to parameter "module_class" of type "(ModelConfiguration) -> Module" in function "__init__"` but perhaps this is related to something else.

Reply:
I did not check but maybe:

`ConformerVGGFrontendV1` actually is like `Callable[[ModelConfiguration], ConformerVGGFrontendV1]`. I'm not sure if it complains because of this difference. You could try `T = TypeVar("T", bound=nn.Module)` and then use `Callable[[ModelConfiguration], T]` as the type here. Or maybe also the same for the cfg, like: `T2 = TypeVar("T2", bound=ModelConfiguration)`, and then `Callable[[T2], T]`. Maybe actually only this second case on cfg.